src/Sylius/Bundle/PromotionBundle/Form/EventListener/SetPriorityFormSubscriber.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\PromotionBundle\Form\EventListener;
+
+use Sylius\Component\Promotion\Model\PromotionInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use Webmozart\Assert\Assert;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+class SetPriorityFormSubscriber implements EventSubscriberInterface
```

Comment:
Shouldn't we inject this behaviour directly to the entity? If position is set to null, use `-1` instead? It isn't anything specific to forms.